components/supervisor/frontend/src/ide/supervisor-service-client.ts

```diff
@@ -49,7 +56,11 @@ export class SupervisorServiceClient {
             }
             if (response.status === 502) {
                 // bad gateway, supervisor is gone
-                return
+                return;
+            }
+            if (response.status === 302 && response.headers.get("location")?.includes("/start/")) {
```

Comment:
do we need to be so precise? maybe `>= 300 && < 400`, i.e. in case of any redirect?